search-commons/src/main/java/no/unit/nva/search2/common/QueryTools.java

```diff
@@ -0,0 +1,235 @@
+        return new Entry<>() {
+            @Override
+            public String getKey() {
+                return strings[0];
+            }
+
+            @Override
+            public String getValue() {
+                return attempt(() -> strings[1]).orElse((f) -> EMPTY_STRING);
+            }
+
+            @Override
+            @JacocoGenerated
+            public String setValue(String value) {
+                return null;
+            }
+        };
+    }
+
+    public static Entry<String, SortOrder> entryToSortEntry(Entry<String, String> entry) {
+        return new Entry<>() {
+            @Override
+            public String getKey() {
+                return entry.getKey();
+            }
+
+            @Override
+            public SortOrder getValue() {
+                var sortOrder = nonNull(entry.getValue()) && !entry.getValue().isEmpty()
+                    ? entry.getValue()
+                    : Defaults.DEFAULT_SORT_ORDER;
+                return SortOrder.fromString(sortOrder);
+            }
+
+            @Override
+            @JacocoGenerated
+            public SortOrder setValue(SortOrder value) {
+                return null;
+            }
+        };
+    }
+
+    static URI nextResultsBySortKey(SwsResponse response, Map<String, String> requestParameter, URI gatewayUri) {
+
+        requestParameter.remove(Words.FROM);
+        var sortedP =
+            response.getSort().stream()
+                .map(Object::toString)
+                .collect(Collectors.joining(COMMA));
+        requestParameter.put(Words.SEARCH_AFTER, sortedP);
+        return fromUri(gatewayUri)
+            .addQueryParameters(requestParameter)
+            .getUri();
+    }
+
+    static String[] splitValues(String value) {
+        return Arrays.stream(value.split(COMMA))
+            .map(String::trim)
+            .toArray(String[]::new);
+    }
+
+    public Stream<Entry<K, QueryBuilder>> queryToEntry(K key, QueryBuilder qb) {
+        final var entry = new Entry<K, QueryBuilder>() {
+            @Override
+            public K getKey() {
+                return key;
+            }
+
+            @Override
+            public QueryBuilder getValue() {
+                return qb;
+            }
+
+            @Override
+            @JacocoGenerated
+            public QueryBuilder setValue(QueryBuilder value) {
+                return null;
+            }
+        };
+        return Stream.of(entry);
+    }
+
+    Stream<QueryBuilder> buildQuery(K key, String... values) {
+        final var searchFields = getSearchFields(key);
+        if (hasMultipleFields(searchFields)) {
+            return Arrays.stream(values)
+                .map(singleValue -> getMultiMatchQueryBuilder(singleValue, searchFields));
+        }
+        return Arrays.stream(values)
+            .map(singleValue -> getMatchQueryBuilder(key, singleValue));
+    }
+
+    Stream<Entry<K, QueryBuilder>> rangeQuery(K key, String value) {
```

Comment:
Many methods in class are missing visibility properties - public/private

Reply:
They are package private